pkg/apis/migration/v1alpha1/model.go

```diff
@@ -161,3 +167,114 @@ func GetSecret(client k8sclient.Client, ref *kapi.ObjectReference) (*kapi.Secret
 
 	return &object, err
 }
+
+// DeleteLabels - removes migration  labels from the resource after restore
+func DeleteLabels(config *rest.Config, uid string) error {
+	GVRs, err := getGVRs(config)
+	if err != nil {
+		return err
+	}
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		MigratedByLabel: uid,
+	}).AsListOptions()
+
+	for _, gvr := range GVRs {
+		list, err := client.Resource(gvr).List(*listOptions)
+		if err != nil {
+			return err
+		}
+		for _, r := range list.Items {
+			labels := r.GetLabels()
+			delete(labels, MigratedByLabel)
+			r.SetLabels(labels)
+			_, err = client.Resource(gvr).Namespace(r.GetNamespace()).Update(&r, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// DeleteMigrated - dispatches delete requests for migrated resources
+func DeleteMigrated(config *rest.Config, uid string) error {
+	GVRs, err := getGVRs(config)
+	if err != nil {
+		return err
+	}
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		MigratedByLabel: uid,
+	}).AsListOptions()
+	foreground := metav1.DeletePropagationForeground
+	deleteOptions := (&k8sclient.DeleteOptions{
+		PropagationPolicy: &foreground,
+	}).AsDeleteOptions()
+
+	for _, gvr := range GVRs {
+		list, err := client.Resource(gvr).List(*listOptions)
+		if err != nil {
+			return err
+		}
+		for _, r := range list.Items {
+			err = client.Resource(gvr).Namespace(r.GetNamespace()).Delete(r.GetName(), deleteOptions)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// WaitForMigratedDeletion - returns true if all of the migrated resources were deleted
+func WaitForMigratedDeletion(config *rest.Config, uid string) (bool, error) {
```

Comment:
Why do we need to wait?

Reply:
The whole purpose of removing the application on the destination side will be reduced to zero, if some of the pods (or other resource interacting with the same location as the application on the source side) on the target side remain running. Also, the second time the migration will run, existing resources won't be updated so the method is needed.